### PR TITLE
Modify rule S5876: Update to LayC format (APPSEC-969) 

### PR DESCRIPTION
**Pull request by sebastien-andrivet-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5876/common/fix/new-session.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+The protection works by ensuring that the session identifier, which is used to identify and track a user's session, is changed or regenerated during the authentication process.
+
+Here's how session fixation protection typically works:
+
+1. When a user visits a website or logs in, a session is created for them. This session is assigned a unique session identifier, which is stored in a cookie or passed through URL parameters.
+
+
+2. In a session fixation attack, an attacker tricks a user into using a predetermined session identifier controlled by the attacker. This allows the attacker to potentially gain unauthorized access to the user's session.
+
+3. To protect against session fixation attacks, session fixation protection mechanisms come into play during the authentication process. When a user successfully authenticates, the session fixation protection mechanism generates a new session identifier for the user's session.
```
Comment (egon-okerman-sonarsource) [inline rules/S5876/common/fix/new-session.adoc:10]:
```suggestion
3. To protect against session fixation attacks, session fixation protection mechanisms come into play during the authentication process. When a user successfully authenticates, this mechanism generates a new session identifier for the user's session.
```
A bit less repetition

**Review by egon-okerman-sonarsource — CHANGES_REQUESTED** [review]
The text is good overall, but I think some adjustments could make it even better. A lot of it is subjective though, so feel free to discuss it in the comments I made.

File: rules/S5876/java/rule.adoc [hunk 2]
```diff
@@ -1,39 +1,29 @@
...
-     .sessionFixation().none(); // Noncompliant: the existing session will continue
-}
-----
+=== Documentation
 
-=== Compliant solution
+include::../common/resources/docs.adoc[]
 
-In a Spring Security's context, session fixation protection can be enabled as follows:
+https://docs.spring.io/spring-security/reference/5.6-SNAPSHOT/servlet/authentication/session-management.html#ns-session-fixation[Session Fixation Attack Protection]
```
[the remark this answers is not in the document]
Comment (sebastien-andrivet-sonarsource) [reply rules/S5876/java/rule.adoc:19]: I have replaced this version-specific link with the non-version-specific one (i.e. latest version).

**Review by egon-okerman-sonarsource — APPROVED** [review]
LGTM 👍

**Pull request merged**